Sources/Context/Context.swift

```diff
@@ -0,0 +1,54 @@
+public protocol Context {
+    static var background: Context { get }
+    static func TODO(_ reason: StaticString?, function: StaticString, file: StaticString, line: UInt) -> Context
```

Comment:
These are subtly wrong in a number of ways:

- It implies I'm working with APIs like `(context: MyLib.Context)` all the time and that context is a concrete type...

Because that's how people will want to use it `myCall(context .TODO())` then, but that's wrong, APIs should try hard to not accept a specific type!

The completion won't work if the parameter is `Context` anyway, because `static func` can not be invoked on protocol, so this: `func c(context: Context); `c(context: .TODO) /*wont compile*/` so this API gives false hope and leads designers into designing the wrong APIs.

❗ This makes it not composable with any other context -- because you implied the concrete type of a library is used in signatures.

Say I'm `HTTPContext` but I need to call CLI with `CLIContext`, every single time I'd have to convert the one context to the other.

This means if I'm `Lambda.Context` I can do `Lambda.Context.TODO("i'm too lazy")` which implies you try to force call sites to look like: `func call(c: Lambda.Context)`

--- 

- The signature is slightly wrong, if it is defined on `protocol Context` it SHOULD return `Self` and not context, right?

Reply:
Yes, it should've. I forgot that returning `Self` doesn't remove the ability to use it as an existential